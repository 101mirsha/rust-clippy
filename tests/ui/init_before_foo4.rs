#![feature(rustc_attrs)]

#[rustc_diagnostic_item = "check_main"]
fn main() {
    if false_value() {
        init();
    }
    foo();
}

#[inline(never)]
fn false_value() -> bool {
    false
}

#[rustc_diagnostic_item = "foo"]
fn foo() {}

#[rustc_diagnostic_item = "init"]
fn init() {}
