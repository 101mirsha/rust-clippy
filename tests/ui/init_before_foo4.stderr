error: call to `foo` not preceded by call to `init`
  --> $DIR/init_before_foo4.rs:8:5
   |
LL |     foo();
   |     ^^^^^
   |
   = note: `#[deny(clippy::init_before_foo)]` on by default

error: aborting due to previous error

